### DISCUSSION TITLE
APP UI Font Change

### DISCUSSION THREAD

**Discussion by tomzx** [damionx7/Zettel-Notes-Documentation#80] [Feature Requests]
Let users choose fonts to be used in UI text. Entire app's text including settings, folders and file names all will follow this selected font.

**Comment by damionx7** [comment]
I have added Roboto and Lato fonts which can be selected in settings (`Settings > Theme > UI Font`) to change Global UI Font.

**Comment by tomzx** [comment]
I'm curious, why can't we use the fonts that can be downloaded (like the editor and viewer fonts under the customize section) and are limited in this case?

I'm asking because I'd like to use a single font across all three (UI, editor, and viewer).

**Reply by damionx7** [reply]
As to change UI font we need `font-family` in styles.xml. And this can be provided only with app. Read more at https://developer.android.com/develop/ui/views/text-and-emoji/fonts-in-xml#font-family.

**Comment by spc789** [comment]
Any chance on adding nerdfonts in the UI / view / edit section ? 
Those are open source fonts patched with over 10000 icons from different sources. (Like powerline/mui etc)

**Reply by damionx7** [reply]
You can always import any font from internal storage for viewer and editor. So, download your favorite font from nerdfonts and just import it in the font selection dialog.

Thanks for informing btw. I didn't know about nerdfonts.

**Reply by spc789** [reply]
Your welcome, Thzre was another reason I lentioned nerdfonts. I was thinking it could be a way to reduce package size using such font (as many of the icons the app uses are ín such a font) 
(Instead if using additional glyphs)

**Reply by damionx7** [reply]
Certainly. I am already use a font file for providing icons in the app, 😄 (Material Fonts). You can see all the provided icons when you change editor toolbar button icon in settings.
